src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+---
+title: Dart 3 migration guide
+description: Tips for how to migrate existing Dart code to be compatible with Dart 3.
+---
+
+Dart 3 is a new major release of Dart, that brings new core capabilities like
+records, patterns, and classes. Alongside these new capabilities the release
+also contains a number of changes, that may break existing code. The present
+document is intended to be a migration guide that may be helpful in resolving
+these.
+
+## Introduction
+
+### Unversioned vs versioned changes
+
+The potentially breaking changes listed below fall into one of two categories:
+
+* **Unversioned changes**: These are changes which are affect any Dart code,
+which is used after upgrading to a Dart 3 SDK. There is no way to "turn off"
+these changes.
+
+* **Versioned changes**: Versioned changes are different, 
+and only apply when the package/app has selected the 
+Dart 3.0 (or above) language version. 
+The language version [is derived](/guides/language/evolution#language-version-numbers)
+from the `sdk` lower-constraint in the `pubspec.yaml` file. An SDK constraint like
+this does *not* apply the Dart 3 versioned changes:
```

Comment:
I'm not sure if this is the _right_ link, but any link to what "selecting the language version" actually entails or how to do it, so... I chose the pubspec guidance? Maybe there's something better.

```suggestion
* **Versioned changes**: These changes only apply when the package or app's
language version is set to >= Dart 3.0. 
The [language version](/guides/language/evolution#language-version-numbers)
is derived from the `sdk` lower-constraint in the
[`pubspec.yaml` file](/guides/packages#creating-a-pubspec). 
An SDK constraint like this does *not* apply the Dart 3 versioned changes:
```